go/log-stressor/log-stressor.go

```diff
@@ -62,4 +69,32 @@ func main() {
             time.Sleep(time.Duration(sleep * float64(time.Second)))
         }
     }
-}
\ No newline at end of file
+}
+
+type formatter func(hash string, messageCount int, payload string)
+
+func NewFormatter(outputFile, outputFormat string) formatter {
+    if outputFile != "" {
+        f, err := os.Create(outputFile)
+        if err != nil {
+            log.Fatalf("Unable to create out file %s: %v", outputFile, err)
+        }
+        log.SetOutput(f)
+    }
+    formatter := FormatForStdOut
+    if outputFormat == "crio" {
```

Comment:
you are calling the format function twice ... change to if/else to save cpu

Reply:
I'm not sure I understand here.  This is a factory function to return a formatter.  It initializes how its output, and defaults the formate function to be STDOUT.  If configured, it will return an alternate function that will be used.  The `NewFormatter` only gets called at start